### ISSUE TITLE
QUIT: On quitting the app ask if to save the script.R

### ISSUE THREAD

**Issue by gjwgit** [gjwgit/rattle#188]
## Description

For the desktop apps there is usually a way to capture the user pressing the Quit/Close button from the window manager. This should popup a dialog to offer to save the SCRIPT into script.R using the same widget as in the SCRIPT feature's SAVE button.

**Comment by zheyxu** [comment]
flutter_window_close: ^1.0.0 is a relevant package but does not work at macOS version.
Check the issue https://github.com/zonble/flutter_window_close/issues/15

**Comment by gjwgit** [comment]
So flutter_window_close works well for windows and linux but not for MacOS apparently.

**Comment by Lutra-Fs** [comment]
I can implement this feature in PR #442, along with the confirmation dialog for closing. Please let me know if you have any thoughts or feedback.

**Comment by zheyxu** [comment]
Thanks @Lutra-Fs, once #442 is merged, I can easily implement this issue. It seems that package **flutter_window_close** is no longer needed. @gjwgit
